pkg/controller/migmigration/validation.go

```diff
@@ -93,6 +95,61 @@ func (r ReconcileMigMigration) validate(ctx context.Context, migration *migapi.M
 		err = liberr.Wrap(err)
 	}
 
+	err = r.validateIntraClusterMigration(ctx, plan, migration)
+	if err != nil {
+		log.V(4).Error(err, "Validation of intra-cluster migration failed")
+		err = liberr.Wrap(err)
+	}
+	return nil
+}
+
+// validateIntraClusterMigration runs validations for intra-cluster migrations
+func (r ReconcileMigMigration) validateIntraClusterMigration(ctx context.Context, plan *migapi.MigPlan, migration *migapi.MigMigration) error {
+	if opentracing.SpanFromContext(ctx) != nil {
+		span, _ := opentracing.StartSpanFromContextWithTracer(ctx, r.tracer, "validateIntraClusterMigration")
+		defer span.Finish()
+	}
+	// this validation only applies to intra-cluster migrations
+	isIntraCluster, err := plan.IsIntraCluster(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	if !isIntraCluster {
+		return nil
+	}
+	conflictingNamespaces := []string{}
+	for srcNs, destNs := range plan.GetNamespaceMapping() {
+		if srcNs == destNs {
+			conflictingNamespaces = append(conflictingNamespaces, srcNs)
```

Comment:
This doesn't account for namespaces in conflict via explicit mapping to other namespaces. It would catch the following:
```
- foo:foo
- bar:baz
```
and it would catch:
```
- foo
- bar
```
But it would not catch:
```
- foo:bar
- bar:foo
```
Instead, I think we need to make sure that there is no intersection in the sets returned by `plan.GetSourceNamespaces()` and `plan.GetDestinationNamespaces()`.

Reply:
@sseago good catch. Thanks for pointing that out

Reply:
@sseago Note that this validation is on MigMigration. I added a new validation to avoid the scenario you mentioned in the Migration Plan instead. the new vlaidation raises a Critical condition on the migration plan making the user correct the mistake right away.